src/ring_theory/unique_factorization_domain.lean

```diff
@@ -942,6 +942,19 @@ quotient.induction_on a $ assume a, decidable.by_cases
 theorem prod_factors [nontrivial α] (s : factor_set α) : s.prod.factors = s :=
 factor_set.unique $ factors_prod _
 
+lemma factors_eq_none_iff_zero [nontrivial α] {a : associates α} :
+  a.factors = option.none ↔ a = 0 :=
+begin
+  split; intro h,
+    { rw [← factors_prod a, factor_set.prod_eq_zero_iff], exact h },
+    { rw h, exact factors_0 }
```

Comment:
```suggestion
⟨λ h, by rwa [← factors_prod a, factor_set.prod_eq_zero_iff], λ h, h.symm ▸ factors_0 ⟩
```
a little term mode 🏌️; `refine` is pretty much a better version of `split`, btw!

Reply:
Nice!